runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -679,8 +754,41 @@ private void createGameTimer(GameTimer timer)
 		infoBoxManager.addInfoBox(t);
 	}
 
+	private void createGameTimer(GameTimer timer)
+	{
+		createGameTimer(timer, true);
+	}
+
 	private void removeGameTimer(GameTimer timer)
 	{
-		infoBoxManager.removeIf(t -> t instanceof TimerTimer && ((TimerTimer) t).getTimer() == timer);
+		removeGameTimer(timer, NO_TIME_PASSED_THRESHOLD);
+	}
+
+	/**
+	 * Removes a rendered GameTimer from the infobox manager. Accepts a `secondsPassedThreshold`
+	 * defining the limit of seconds in which to allow the removal of a timer.
+	 *
+	 * @param timer           			The timer to be added to the infobox manager.
+	 * @param secondsPassedThreshold 	Seconds allowed to be elapsed since start of timer in
+	 *                                  order to be removed. Mainly used to allow graphics to
+	 *                                  filter freeze timers on first interaction between various
+	 *                                  events.
+	 *
+	 *                                  i.e.: default freeze timer is BARRAGE upon
+	 *                                  frozen message, but graphic indicates BLITZ. This
+	 *                                  threshold allows for the removal of BARRAGE. On subsequent
+	 *                                  freeze spell casts the number of seconds passed should
+	 *                                  be greater than this threshold and should not allow the
+	 *                                  current freeze timer to be removed, as it is in effect.
+	 */
+	private void removeGameTimer(GameTimer timer, long secondsPassedThreshold)
```

Comment:
`long secondPassedThreshold` may be better off being `Duration timePassedThreshold`, but I think that's minor enough not to matter at the moment.